src/main/java/com/aws/greengrass/clientdevices/auth/metrics/MetricsEmitter.java

```diff
@@ -5,27 +5,45 @@
 
 package com.aws.greengrass.clientdevices.auth.metrics;
 
+import com.aws.greengrass.builtin.services.telemetry.ComponentMetricIPCEventStreamAgent;
+import lombok.Getter;
+import software.amazon.awssdk.aws.greengrass.GeneratedAbstractPutComponentMetricOperationHandler;
+import software.amazon.awssdk.aws.greengrass.model.PutComponentMetricRequest;
+import software.amazon.awssdk.aws.greengrass.model.PutComponentMetricResponse;
+import software.amazon.awssdk.aws.greengrass.model.UnauthorizedError;
+import software.amazon.awssdk.eventstreamrpc.OperationContinuationHandlerContext;
+
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.ScheduledFuture;
 import java.util.concurrent.TimeUnit;
 import javax.inject.Inject;
 
 public class MetricsEmitter {
+    @Getter
     private ScheduledFuture<?> future;
+    private final ComponentMetricIPCEventStreamAgent componentMetricIPCAgent;
+    private final OperationContinuationHandlerContext context;
     private final ScheduledExecutorService ses;
     private final Object emitMetricsLock = new Object();
     private final ClientDeviceAuthMetrics metrics;
+    private boolean unauthorized = false;
 
     /**
      * Constructor for the MetricsEmitter.
      *
-     * @param ses     {@link ScheduledExecutorService}
-     * @param metrics {@link ClientDeviceAuthMetrics}
+     * @param ses                     {@link ScheduledExecutorService}
+     * @param metrics                 {@link ClientDeviceAuthMetrics}
+     * @param componentMetricIPCAgent Component Metric IPC Event Stream Agent
+     * @param context                 Operation continuation handler
      */
     @Inject
-    public MetricsEmitter(ScheduledExecutorService ses, ClientDeviceAuthMetrics metrics) {
+    public MetricsEmitter(ScheduledExecutorService ses, ClientDeviceAuthMetrics metrics,
+                          ComponentMetricIPCEventStreamAgent componentMetricIPCAgent,
+                          OperationContinuationHandlerContext context) {
```

Comment:
you cannot inject an `OperationContinuationHandler`. You must make one for _every_ request.